### PR TITLE
MoH-17-Add-donation-Show-high-low-value-after-selecting-item

### PR DESCRIPTION
**Pull request by PhuHoang177**
# Description
- Instantly show the values of High/Low (in drop-down options) only when: Both item and category must be selected AND the value of NewOrUsed is "Used."
- If the values of High/Low are already showed and selected, but then user changes the item AND the category (both must be selected), the new values of High/Low are also automatically updated and showed.
- If the user changes value of NewOrUsed to "New," the field High/Low will be disabled without affecting the values of High/Low.

## Relevant issue(s)
- https://github.com/hack4impact-utk/Maintenance-Team/issues/17

## Testing result(s):
- https://jam.dev/c/2ea750cc-2a08-493f-9ed5-e206d52db09b

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have assigned reviewers to this PR

### CODE REVIEW

File: src/components/donationItemForm/index.tsx [hunk 1]
```diff
@@ -40,6 +40,23 @@ export default function DonationItemForm({
...
+  // Checking undefined, null, and emptiness before show the Higg/Low values
+  const updateHighLowVals = (newOrUsed: string, itemRes: ItemResponse) => {
+    if (!!newOrUsed) {
+      if (newOrUsed === 'Used') {
+        if (!!itemRes) {
+          setHighLowVals([`High ($${itemRes.high})`, `Low ($${itemRes.low})`]);
+        }
+      }
+    }
+  };
```
Comment (andlrutt) [inline src/components/donationItemForm/index.tsx:58]:
```suggestion
  // Checking undefined, null, and emptiness before show the High/Low values
  const updateHighLowVals = (newOrUsed: string, itemRes: ItemResponse) => {
    if (newOrUsed === 'Used' && !!itemRes) {
      setHighLowVals([`High ($${itemRes.high})`, `Low ($${itemRes.low})`]);
    }
  };
```

File: src/components/donationItemForm/index.tsx [hunk 2]
```diff
@@ -155,14 +175,13 @@ export default function DonationItemForm({
...
+
       <Grid item xs={12} sm={4}>
         <FormControl
           fullWidth
           disabled={donationItemData.newOrUsed !== 'Used' || disabled} // Disable if new
         >
-          <InputLabel id="high-or-low-value-label">
-            High or Low Value?
-          </InputLabel>
+          <InputLabel id="high-or-low-value-label">High or Low?</InputLabel>
```
Comment (andlrutt) [inline src/components/donationItemForm/index.tsx:184]: I'd lean towards keeping this as "High or Low Value?" That way it's clear to the user that they are entering in an item evaluation. Open to discussion!

**Review by andlrutt — CHANGES_REQUESTED** [review]
Very creative solution, I like it!

Is it possible to clear out the evaluation values when an item/category is edited or emptied? As soon as that happens, it means that the evaluation values are no longer current/relevant. If it's a big undertaking or would make this story extend beyond our sprint (ends thursday), we can alternatively make a follow-up story no worries, just let me know.

File: src/components/donationItemForm/index.tsx (hunk 1 above)
Comment (andlrutt) [reply src/components/donationItemForm/index.tsx:58]: And correct me if I'm wrong, but `itemRes` is not nullable, so we should _always_ have a value there, yes? May be able to remove `!!itemRes` as well. Typescript shouuuuld ensure it. Double check me though.

File: src/components/donationItemForm/index.tsx (hunk 2 above)
Comment (PhuHoang177) [reply src/components/donationItemForm/index.tsx:184]: It was my bad that I edit that block of code, then later not fully revert back to the original code. We should keep the original ("**High or Low Value?**").

File: src/components/donationItemForm/index.tsx (hunk 1 above)
Comment (PhuHoang177) [reply src/components/donationItemForm/index.tsx]: You are correct, I removed all the undefined/null evaluation and it still works fine.

**Review by andlrutt — DISMISSED** [review]

**Review by andlrutt — APPROVED** [review]

**Pull request merged**